tests/common/helpers/platform_api/scripts/platform_api_server.py

```diff
@@ -7,7 +7,7 @@
 
 from BaseHTTPServer import HTTPServer, BaseHTTPRequestHandler
 
-import sonic_platform
+import sonic_platform.platform
```

Comment:
This should not be necessary. On line 14 it is referenced by the parent. What caused you to think this change is necessary?

Reply:
I noticed that platform_api_server.py failed to run in pmon container because an Exception is thrown:
```
Traceback (most recent call last):
  File "/opt/platform_api_server.py", line 14, in <module>
    platform = sonic_platform.platform.Platform()
AttributeError: 'module' object has no attribute 'platform'
```

That's why I change it.

Reply:
Interesting. Which platform did you see this on?

Reply:
The platform I saw this problem on is as following

SONiC Software Version: SONiC.20191130.40

Distribution: Debian 9.12
Kernel: 4.9.0-11-2-amd64
Build commit: 289a32803
Build date: Mon Jun 29 11:39:04 UTC 2020
And I run this test on Arista-7260cx3.

 I also run on a msn-2700, and have the same issue. I think the reason for the problem is we mount a path with a empty __init__.py to the container.
"Source": "/usr/lib/python2.7/dist-packages/arista/utils/sonic_platform",
"Destination": "/usr/lib/python2.7/dist-packages/sonic_platform",

Reply:
This is strange. I tested against a master branch image running on a MSN-2700 and didn't see this issue. Could you please install the latest public master image from Jenkins on your MSN-2700 and report back whether or not you see the failure?

Reply:
Sure. I will have it tested when the DUT is ready.

Reply:
@bingwang-ms: I just ran the platform tests against a DuT running SONiC.20191130.40, and I did not see the issue you reported. Can you try updating your mgmt repo to the latest code?

Reply:
@jleveque That's interesting. We are running the same SONiC version SONiC.20191130.40. I don't think the issue is cause by sonic-mgmt, as I loggined into the pmon container, and got the same error in a python shell.
```
root@str-7260cx3-acs-1:/# python
Python 2.7.13 (default, Sep 26 2018, 18:42:22) 
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sonic_platform
>>> dir(sonic_platform)
['__builtins__', '__doc__', '__file__', '__name__', '__package__', '__path__']
>>> sonic_platform.__file__
'/usr/lib/python2.7/dist-packages/sonic_platform/__init__.pyc'
>>> 
```

We could see that the dir command returns a list without any internal package.  Could you check if the **/usr/lib/python2.7/dist-packages/sonic_platform/__init__.py**  file is empty?

Reply:
Yes, the `__init__.py` file is empty, which should not be an issue. The presence of an `__init__.py` file only signifies to Python the existence of a package; it is not required to have any content. I think you're looking at a red herring here.

Reply:
So, could you run 
```
import sonic_platform
sonic_platform.platform.Platform()
```
in a python shell in the pmon container?
I always get the following error:
```
root@str-7260cx3-acs-1:/# python
Python 2.7.13 (default, Sep 26 2018, 18:42:22) 
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sonic_platform
>>> platform = sonic_platform.platform.Platform()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'platform'
>>> 
```

Reply:
Yes. Just worked fine for me on SONiC.20191130.40, MSN-2700:

```
admin@sonic:~$ docker exec -it pmon bash
root@sonic:/# python
Python 2.7.13 (default, Sep 26 2018, 18:42:22) 
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sonic_platform
>>> pform = sonic_platform.platform.Platform()
>>> chas = pfprm.get_chassis()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'pfprm' is not defined
>>> chas = pform.get_chassis() 
>>> chas.get_system_eeprom_info()
{'0xFE': '0xA4B42DB0', '0x28': 'x86_64-mlnx_x86-r0', '0x29': '2016.11-5.1.0008-9600', '0x22': 'MSN2700-CS2FO', ...
>>> exit()
```

Reply:
That's interesting! I tested it on a MSN-2700 and it works now. But the Arista-7260 with exactly same SONiC version (SONiC.20191130.40) still not works.
The gap I found between these two platform is:
On Arista-7260, the **/usr/lib/python2.7/dist-packages/sonic_platform** is mount into the container:
```
admin@str-7260cx3-acs-1:~$ docker inspect pmon | grep dist
                "/usr/lib/python2.7/dist-packages/arista:/usr/lib/python2.7/dist-packages/arista:ro",
                "/usr/lib/python2.7/dist-packages/arista/utils/sonic_platform:/usr/lib/python2.7/dist-packages/sonic_platform:ro",
                "Source": "/usr/lib/python2.7/dist-packages/arista",
                "Destination": "/usr/lib/python2.7/dist-packages/arista",
                "Source": "/usr/lib/python2.7/dist-packages/arista/utils/sonic_platform",
                "Destination": "/usr/lib/python2.7/dist-packages/sonic_platform",
```

But I can't see the similar mount on MSN-2700. And the package sonic_platform in pmon container on MSN2700 is located in  **/usr/local/lib/python2.7/dist-packages/sonic_platform** , which is different to Arista-7260(**/usr/lib/python2.7/dist-packages/sonic_platform**). 
```
root@str-msn2700-02:/# python
Python 2.7.13 (default, Sep 26 2018, 18:42:22) 
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sonic_platform
>>> pform = sonic_platform.platform.Platform()
>>> dir(sonic_platform)
['__all__', '__builtins__', '__doc__', '__file__', '__name__', '__package__', '__path__', 'chassis', 'eeprom', 'fan', 'platform', 'psu', 'thermal']
>>> sonic_platform.__file__
'/usr/local/lib/python2.7/dist-packages/sonic_platform/__init__.pyc'
``` 
 Will this gap leads to ths issue?

Reply:
This difference should not cause the issue. These are both valid Python paths. I believe there is an issue with the Arista sonic_platform implementation.

Reply:
@jleveque this issue should have been solved by https://github.com/Azure/sonic-buildimage/pull/5408 which merged on September 18. According to the logs provided in this issue, the image used by this testing predates the fix.
For some reason I don't recall seeing a notification for this issue. We ended up figuring this out while working on more platform API features.